packages/teuchos/parameterlist/test/yaml/YamlParameterList.cpp

```diff
@@ -281,8 +281,14 @@ namespace TeuchosTests
       " small number: 54\n"
       " big number: 72057594037927936\n");
     TEST_EQUALITY(pl->isType<int>("small number"), true);
+#ifdef HAVE_TEUCHOSPARAMETERLIST_YAMLCPP
+    TEST_EQUALITY(pl->isType<double>("big number"), true);
+    long long big_number = static_cast<long long>(pl->get<double>("big number"));
+    TEST_EQUALITY(big_number, 72057594037927936ll);
+#else
```

Comment:
YAML only recognizes int, double, or string datatypes, meaning yaml-cpp has to parse this large number as a `double` instead of a `long long`.

Reply:
This comment is the only thing that makes me a uneasy. The parameter list supports "long long int" and from the looks of this test, our internal yaml parser could handle 64bit ints (maybe writing an illegal yaml file?). But now moving to yaml-cpp means we can no longer support "long long" data types directory when reading and writing to yaml. We might want to look into a better way to read and write long long entries to yaml so that the plist generated from a yaml file has the correct data type of long long instead of forcing everyone to use double.